### DISCUSSION TITLE
Project financial backing

### DISCUSSION THREAD

**Discussion by ThibFrgsGmz** [nasa/fprime#1191] [Ideas]
Have you considered starting a crowdfunding collective to help fund the upkeep of the project?

If they don't have computer skills, I imagine that some members of the general public would like to be able to contribute financially to the project.

Contributions could be made on a regular or irregular basis by organizations (private companies) or individuals.

One can imagine that the collected budget could be used to finance different things, such as :

- adding features, for example by a trainee that you supervise ;
- financing source code audits, by accredited organizations or, for fun, by youtubers who would also make videos;
- contests for bug hunters or critical features hunters (I'm referring to you, https://github.com/nasa/fprime/issues/221 :) ) ;
- funding for targeted advertising campaigns.

As a funding platform, I'm thinking of Open Collective that Facebook's Research Unit uses for some of its projects: [Docusaurus contributors](https://github.com/facebook/docusaurus#backers).

**Comment by astroesteban** [comment]
Pretty cool! I see that a large project like [curl](https://github.com/curl/curl) uses it too. 

Given that F' is under the NASA organization account does that impact the project's ability to adopt something like this?

**Reply by LeStarch** [reply]
I think being under the NASA account would impede these efforts, but I will ask around.

**Reply by LeStarch** [reply]
In short, we will look into this but it would likely take some time to sort out.  I wouldn't expect anything particularly soon on this regard.

**Reply by ThibFrgsGmz** [reply]
I suppose it's difficult for you to request a budget for anything related to project marketing/communication; this could be a way to fund that.

One can imagine crowdfunding campaigns for features and goodies for backers, but the best platform seems to be [KickStarter](https://www.kickstarter.com/).

**Comment by LeStarch** [comment]
For those with typing skills, we can always use documentation write-ups and corrections:

https://github.com/nasa/fprime/issues?q=is%3Aissue+is%3Aopen+label%3ADocumentation

This can be done directly on Github too and I'll do my best to help integration anything delivered!

**Comment by ThibFrgsGmz** [comment]
What resources are available to companies (especially non-U.S. companies) considering using FPrime to support the project over time and ensure its sustainability?

Is some sort of international association conceivable? Like the Rust Foundation or the RISCV Foundation?
